File: articles/distribution-shift-and-model-decay/prolog/decay_risk.pl
decay_risk(InputDrift, LabelDrift, PerformanceDecay, CalibrationGap, SubgroupGap, OverrideRate, Score) :-
    Score is (InputDrift + LabelDrift + PerformanceDecay + CalibrationGap + SubgroupGap + OverrideRate) / 6.

% Query example:
% ?- decay_risk(0.31, 0.16, 0.10, 0.14, 0.15, 0.11, Score).

File: articles/distribution-shift-and-model-decay/haskell/DecayRisk.hs
module Main where

main :: IO ()
main = do
  let inputDrift = 0.31
      labelDrift = 0.16
      performanceDecay = 0.10
      calibrationGap = 0.14
      subgroupGap = 0.15
      overrideRate = 0.11
      score = (inputDrift + labelDrift + performanceDecay + calibrationGap + subgroupGap + overrideRate) / 6
  putStrLn ("decay_risk_score=" ++ show score)
